### DISCUSSION TITLE
Not respecting the rules

### DISCUSSION THREAD

**Discussion by ILKAY-BRAHIM** [geeksblabla/blanat#306] [General]
It seems there's an issue with the participants' output, displaying 7 lines instead of the required 6, including an extra empty line.

**Comment by aboullaite** [comment]
Who you are referring to ?

**Reply by ILKAY-BRAHIM** [reply]
the majority of leaderboard

**Reply by aboullaite** [reply]
Would be helpful to put names :) we will take a look

**Reply by yousfiSaad** [reply]
I wonder how the results would be if we run the programs with a different set of cities/products. after all, the assumption is that the maximum number of the items is fixed not the set of items set itself.
Concerning the issue that Mr @ILKAY-BRAHIM pointed, I don't see that's fair to penalize participants for this issue, since the outputs are accepted during the competition.
`Be careful when you format the string that you will print, the spaces have to be respected, you will receive a Wrong answer even if the data is correct. ` this citation from readme implicitly says that an empty line at the end is not a big deal

**Reply by ILKAY-BRAHIM** [reply]
@aboullaite  this names and am not test all code of users 
NAME               | Error
-----------------------------
superma-C           | 7 line 
k34n4y138           | 7 line 
anassajaanan       | 7 line 
ibrataha8           | 7 line
viierr              | 7 line 
yousfiSaad           | 7 line
sqrt-minus-one     | 7 line
dauom              | 7 line + false result
webNeat            | 7 line
ouakki.             | 7line
samir              | 7 line

**Reply by aboullaite** [reply]
If its an empty line its not a big deal as mentionned above. The runner takes care of it as long as the city followed by 5 cheapest product is printed. Please let us know if you have any further concerns.

@yousfiSaad I see what you mean. But it is part of the challenge, and we stated from the early days that the list of cities and product is fixed with the aim on focusing on the reading the file. That is a fair optimisation given the constraints and it was hard to change it mid challenge.

**Reply by yousfiSaad** [reply]
I was tempted to use gperf, but I didn't for two reasons, 
1. I wanted to let my program handle different sets of items, (I'm sure Adnan will add a product to his list in the future)
2. the custom map that I created gave me pretty good performances when tested on my machine.

**Reply by ILKAY-BRAHIM** [reply]
But in format of output its required 6 lines
And the some of them its wrong answer

**Reply by chermehdi** [reply]
the runner will trim trailing spaces before comparing outputs, just to give people a bit of flexibility I wouldn't consider printing an extra line a bad thing / wrong answer. I have did competitive programming for a while, no one penalizes solutions for that extra line, and I don't think we should either.

**Comment by ILKAY-BRAHIM** [comment]
@yousfiSaad  I asked before if cities and product will remain stable, and it was confirmed.

**Comment by k34n4y138** [comment]
what @ILKAY-BRAHIM means is the trailing \n after eachline,
is the 6th line required to terminated by EOF or is it okay to have \n as last byte
the majority of the leaderboard prints 6th line with trailing nl including me

**Comment by yousfiSaad** [comment]
Just for information:
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html
POSIX considers lines (at the end of a file) that don't end with `\n` to be `incomplete lines`

I don't like files without \n char at the end, I guess that makes me POSIX compliant 😄

**Reply by ILKAY-BRAHIM** [reply]
Thanks for the clarification

**Reply by yousfiSaad** [reply]
You're welcome

**Comment by samir-** [comment]
Sorry to be a bit personal here. I think the "issue" you reported @ILKAY-BRAHIM was not sincere/honest ,especially reported after the finish while you knew it before and you likely wanted to leverage it as a winning factor for your own benefit. You targeted all users above you in the cur leaderboard ;) . Not to mention that all sols above you had this issue and you know that the level of implementation exceeds largely these type of "issues".

**Reply by ILKAY-BRAHIM** [reply]
@samir-  But the lederbord is temporary its mean if test manuly and find the wrong answer wil be removd in lederbord
Hadshu kay3ni ya 7aydo kolshu 7ta 3la new line wla ikhaliw kolshu wa5a 3andhoom resul ghalta

**Reply by ILKAY-BRAHIM** [reply]
@samir-  By the way all users have a newline and of corse i test the best in the lederbord first

**Reply by samir-** [reply]
I don't think you got my point. And you didn't adresse my point about the cheap shot. I will leave it here :)

**Reply by aboullaite** [reply]
Just in case I didn't make myself clear from my first comment. A new line is not counted as incorrect answer and is not eliminatory. The further verification / validation will concern all the actual leader board based on correctness, speed and (valid) optimisations. 

Best of luck to all of you
